Feature/allow additional handlers

Pudge601 opened the pull request:
Fixes #5

Commit:
Use dynamic method names for creating loggers in the factory

This allows for easier extensibility of the factory to add new logger types by extending the factory class and adding new ceate_Logger methods.

Fixes #5

Commit:
Remove the logger type constants and all uses thereof

Logger type names are considered a lot more variable now that we're determining the logger creation method dynamically, and now that new logger
types can be easily added through extending the factory class.

This makes the use of constants for logger type names seem wrong, so just use literal strings instead.

#5

chrisminett commented on src/Factory.php:
```diff
@@ -26,20 +22,15 @@ public function createLogger($name, array $config)
-                throw new \DomainException(sprintf('Cannot find a logger type named "%s"', $type));
+        $type = trim($config['type']);
+        if (!$type) {
+            throw new \DomainException(sprintf('Logger type cannot be empty', $type));
```
The `$type` param has no-where to be injected.

Commit:
Remove call to sprintf

We're not interpolating any variables into this error message, so no need to put the message through sprintf at all.

#5
